### PR TITLE
First version

### PR DESCRIPTION
**Pull request by alanhoff**

### CODE REVIEW

**Comment by alanhoff** [comment]
@bsiddiqui ready for review

File: index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  let modelPrototype = bookshelf.Model.prototype
+
+  // Extends the default model class
+  bookshelf.Model = bookshelf.Model.extend({
+    initialize: function (attributes, options) {
+      modelPrototype.initialize.call(this)
+
+      if (this.uuid) {
+        this.defaults = merge({
+          [this.idAttribute]: typeof settings.type === 'string'
```
Comment (bsiddiqui) [inline index.js:23]: when would `settings.type !== 'string'` since you're providing defaults?

Comment (bsiddiqui) [inline index.js:23]: oh nvm custom generator

**Comment by bsiddiqui** [comment]
LGTM

**Pull request merged**